AllReadyApp/Web-App/AllReady/Areas/Admin/Controllers/TenantController.cs

```diff
@@ -1,43 +1,41 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Threading.Tasks;
-using Microsoft.AspNet.Mvc;
+﻿using Microsoft.AspNet.Mvc;
 using AllReady.Models;
-using Microsoft.Data.Entity;
 using Microsoft.AspNet.Authorization;
-
-// For more information on enabling MVC for empty projects, visit http://go.microsoft.com/fwlink/?LinkID=397860
+using MediatR;
+using AllReady.Areas.Admin.Features.Tenants;
+using AllReady.Areas.Admin.Models;
 
 namespace AllReady.Areas.Admin.Controllers
 {
-    //[Area("TenantAdmin")]
     [Area("Admin")]
     [Authorize("TenantAdmin")]
     public class TenantController : Controller
     {
+        private readonly IMediator _bus;
         private readonly IAllReadyDataAccess _dataAccess;
 
-        public TenantController(IAllReadyDataAccess dataAccess)
+        public TenantController(IMediator bus, IAllReadyDataAccess dataAccess)
         {
+            _bus = bus;
             _dataAccess = dataAccess;
         }
 
         // GET: Tenant
-        public async Task<IActionResult> Index()
+        public IActionResult Index()
         {
-            return await Task.Run(() => View(_dataAccess.Tenants));
+            var list = _bus.Send(new TenantListQuery());
+            return View(list);
         }
 
         // GET: Tenant/Details/5
-        public IActionResult Details(System.Int32? id)
+        public IActionResult Details(int? id)
```

Comment:
No need for nullable id here. Change this to int and then we can delete the check for null after.